src/auth/googleauth.ts

```diff
@@ -634,4 +651,53 @@ export class GoogleAuth {
           if (callback) callback(null, body);
         });
   }
+
+
+  /**
+   * Returns the contents of the metadata of GC instance
+   * @return object representation of the service account JSON
+   */
+  getCredentials(
+      callback?: (err: Error|null, credentials?: CredentialBody) => void) {
+    this._checkIsGCE((err, gce) => {
+      if (gce) {
+        // For GCE, return the service account details from the metadata server
+        this.transporter.request(
+            {
+              method: 'GET',
+              uri:
+                  'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/?recursive=true',
+              headers: {'Metadata-Flavor': 'Google'}
+            },
+            (err, body, res) => {
+              if (err || !res || res.statusCode !== 200 || !body) {
+                return;
+              } else {
+                // Callback with the body
+                const credential:
+                    CredentialBody = {client_email: body['default']['email']};
```

Comment:
Any reason to not use `body.default.email`? What happens when `body.default` or `email` is `undefined`?

Reply:
Changed to body.default.email. Added check if body.default or body.default.email is undefined.